lib/builder/builder.js

```diff
@@ -254,12 +254,20 @@ export default class Builder extends Tapable {
       templateVars.router.routes = this.options.build.createRoutes(this.options.srcDir)
     }
 
-    await this.applyPluginsAsync('extendRoutes', { routes: templateVars.router.routes, templateVars, r })
+    const extendedRoutes = await this.applyPluginsAsync('extendRoutes', { routes: templateVars.router.routes, templateVars, r })
```

Comment:
I think this variable is always `undefined` as tapable plugins are not meant to return things.